src/Common/hooks/useNotificationSubscribe.ts

```diff
@@ -0,0 +1,131 @@
+import { useDispatch } from "react-redux";
+import {
+  getUserPnconfig,
+  updateUserPnconfig,
+  getPublicKey,
+} from "../../Redux/actions";
+import { useState } from "react";
+import * as Sentry from "@sentry/browser";
+import { useTranslation } from "react-i18next";
+import { Error } from "../../Utils/Notifications.js";
+import useAuthUser from "../../Common/hooks/useAuthUser";
+
+export default function useNotificationSubscribe() {
+  const { username } = useAuthUser();
+  const dispatch: any = useDispatch();
+  const [isSubscribed, setIsSubscribed] = useState("");
+  const [isSubscribing, setIsSubscribing] = useState(false);
+  const { t } = useTranslation();
+
+  const intialSubscriptionState = async () => {
+    try {
+      const res = await dispatch(getUserPnconfig({ username: username }));
+      const reg = await navigator.serviceWorker.ready;
+      const subscription = await reg.pushManager.getSubscription();
+      if (!subscription && !res?.data?.pf_endpoint) {
+        setIsSubscribed("NotSubscribed");
+      } else if (subscription?.endpoint === res?.data?.pf_endpoint) {
+        setIsSubscribed("SubscribedOnThisDevice");
+      } else {
+        setIsSubscribed("SubscribedOnAnotherDevice");
+      }
+    } catch (error) {
+      Sentry.captureException(error);
+    }
+  };
```

Comment:
I assume we are exporting this function and using it inside use effect in the importing file, Why not directly do it here?